src/extension/api.ts

```diff
@@ -159,6 +160,14 @@ export interface Windows extends Observable<Window[]> {
      * @returns The user's input, or null if the user (or the client) canceled the input request.
      */
     showInputBox(message: string, defaultValue?: string): Promise<string | null>
+
+    /**
+     * Sets the decorations for the given document. All previous decorations for the document are cleared.
+     *
+     * @param resource The document to decorate.
+     * @param decorations The decorations to apply to the document.
+     */
+    setDecorations(resource: TextDocumentIdentifier, decorations: TextDocumentDecoration[]): void
```

Comment:
I imagine each extension would prefer its decorations to be independent of other extensions (so that when one extension clears its decorations, the other extensions' decorations are still present).